libraries/lib-cloud-audiocom/UserService.cpp

```diff
@@ -0,0 +1,229 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  UserService.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+#include "UserService.h"
+
+#include <memory>
+#include <vector>
+
+#include <wx/file.h>
+
+#include <rapidjson/document.h>
+
+#include "ServiceConfig.h"
+#include "OAuthService.h"
+
+#include "BasicUI.h"
+#include "FileNames.h"
+#include "Observer.h"
+#include "Prefs.h"
+
+#include "IResponse.h"
+#include "NetworkManager.h"
+#include "Request.h"
+
+#include "CodeConversions.h"
+
+namespace cloud::audiocom
+{
+namespace
+{
+wxString MakeAvatarPath()
+{
+   const wxFileName avatarFileName(FileNames::ConfigDir(), "avatar");
+   return avatarFileName.GetFullPath();
+}
+   
+StringSetting userName { L"/cloud/audiocom/userName", "" };
+StringSetting displayName { L"/cloud/audiocom/displayName", "" };
+StringSetting avatarEtag { L"/cloud/audiocom/avatarEtag", "" };
+
+Observer::Subscription authStateChangedSubscription =
+   GetOAuthService().Subscribe(
+      [](const auto& state)
+      {
+         if (state.authorised)
+            GetUserService().UpdateUserData();
+         else
+            GetUserService().ClearUserData();
+      });
+
+} // namespace
+
+void UserService::UpdateUserData()
+{
+   auto& oauthService = GetOAuthService();
+
+   if (!oauthService.HasAccessToken())
+      return;
+
+   using namespace audacity::network_manager;
+
+   Request request(GetServiceConfig().GetAPIUrl("/me"));
+   request.setHeader("Authorization", std::string(oauthService.GetAccessToken()));
```

Comment:
Perhaps the string literals given to `setHeader` should be named constants to avoid repetition of them within this new library.